### DISCUSSION TITLE
How run a custom command without sourcing a file

### DISCUSSION THREAD

**Discussion by LinArcX** [nushell/nushell#7814] [Q&A] [answered]
Imagine I have a file called `test.nu` that contains:
```
def "foo" [name] {
  $"hello from foo --> ($name)"
}

def "bar" [name] {
  $"hello from bar --> ($name)"
}

def "zoo" [name] {
  $"hello from zoo --> ($name)"
}
```

I want to run each command without sourcing the file, like this:
`nu test.nu zoo`

How can I do that?

Edit: In bash scripts, I used this trick to achieve the same goal:
```
foo() {
    echo ">>> foo"
}
bar() {
    echo ">>> bar"
}
zoo() {
    echo ">>> zoo"
}

case "$1" in
    "") ;;
    foo) "$@"; exit;;
    bar) "$@"; exit;;
    zoo) "$@"; exit;;
    *) log_error "Unkown function: $1()"; exit 2;;
esac
```

**Comment by fdncred** [comment, marked as answer]
That functionality does not exist yet.  I believe the only function that can be run like that is a function called `main`. You may see if you can create a function named main with a ...rest arg and see if you can do anything with those parameters. i'm not sure.

**Reply by LinArcX** [reply]
Yes, exactly. `main` function is what I wanted. Thanks.
